flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/FlinkDistribution.java

```diff
@@ -85,24 +86,31 @@ public FlinkDistribution() {
 			Assert.fail("The distDir property was not set. You can set it when running maven via -DdistDir=<path> .");
 		}
 		final String backupDirProperty = System.getProperty("logBackupDir");
-		logBackupDir = backupDirProperty == null
-			? Optional.empty()
-			: Optional.of(Paths.get(backupDirProperty));
-		final Path flinkDir = Paths.get(distDirProperty);
+		logBackupDir = backupDirProperty == null ? null : Paths.get(backupDirProperty);
+		originalFlinkDir = Paths.get(distDirProperty);
+	}
+
+	@Override
+	public void before() throws IOException {
+		temporaryFolder.create();
+
+		final Path flinkDir = temporaryFolder.newFolder().toPath();
+
+		LOG.info("Copying distribution to {}.", flinkDir);
+		AutoClosableProcess.create(
```

Comment:
That's right.   In my macOsx, the FileUtils.copyDirectory only copy the file content but lost the attribution,   so the file have no +x  ACL, the command such as  <flink-dist-dir>/bin/jobmanager.sh will be failure.

Reply:
To make the code less platform dependent, could you try a version based on `Files.walkFileTree()` and `Files.copy` with the `COPY_ATTRIBUTES` `CopyOption`?

Reply:
Ya, Sounds great.  I see there's a  copyDirectory method in AbstractDownloadCache,  I think can move it to TestUtil, so that both the AbstractDownloadCache and FlinkDistribution can share it.  Thanks.